server/src/main/java/org/apache/druid/guice/DruidBinders.java

```diff
@@ -62,32 +66,68 @@ public static MapBinder<Class<? extends Query>, QueryLogic> queryLogicBinderType
     );
   }
 
-  public static QueryLogicBinder queryLogicBinder(Binder binder)
+  public static QueryBinder queryBinder(Binder binder)
   {
-    return new QueryLogicBinder(binder);
+    return new QueryBinder(binder);
   }
 
-  public static class QueryLogicBinder
+  public static class QueryBinder
   {
-    private MapBinder<Class<? extends Query>, QueryLogic> queryLogicMapBinder;
-    private Binder binder;
+    MapBinderHelper<Class<? extends Query>, QueryLogic> queryLogicBinder;
+    MapBinderHelper<Class<? extends Query>, QueryRunnerFactory> queryRunnerFactoryBinder;
+    MapBinderHelper<Class<? extends Query>, QueryToolChest> queryToolChestBinder;
+
 
-    public QueryLogicBinder(Binder binder)
+    public QueryBinder(Binder binder)
     {
-      this.binder = binder;
-      queryLogicMapBinder = DruidBinders.queryLogicBinderType(binder);
+      queryLogicBinder = new MapBinderHelper<>(
+          binder, queryLogicBinderType(binder), ImmutableSet.of(LazySingleton.class)
```

Comment:
Taking the list of scopes is not really helpful here.  There should just be a `naiveBinding` method for each of the types of things that forces into a `LazySingleton`.  If someone wants something other than that, you need to make both the `MapBinder` and the `Binder` available to them through getters.  Exposing the MapBinder and Binder makes this object a true helper because it doesn't inhibit users from doing what they need.  If you hide either of those objects, you are now artificially hiding parts of Guice from someone who uses this object.

Reply:
I don't think anyone will need that - this class pushes the contract that all `queryX` classes must be `LazySingleton` -s.

Why shouldn't all `queryX` be `LazySingleton`-s?

Since `DefaultQueryRunnerFactoryConglomerate` will pick up all these and hand it out to other components in the system - I'm not sure if we need `LazySingleton` here..

Reply:
none of the changes in this file are really necessary anymore - as the PR now relies on using `QueryRunnerFactoryModule` to load the all these things.

should I undo the change in this file?